src/components/Calculator.js

```diff
@@ -1,59 +1,173 @@
-import calculate from '../logic/calculate';
-import Button from './Button';
-
-class Calculator extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      calcObj: {
-        total: null,
-        next: null,
-        operation: null,
-      },
-    };
-    this.onClick = this.onClick.bind(this);
-  }
-
-  onClick(buttonName) {
-    const { calcObj } = this.state;
-    const newCalcObj = { ...calcObj, ...calculate(calcObj, buttonName) };
-    this.setState({
-      calcObj: newCalcObj,
-    });
-  }
-
-  render() {
-    const { calcObj } = this.state;
-    const result = calcObj.next || calcObj.total;
-    return (
-
-      <div className="calculator-wrapper">
-        <div className="result-section">
-          {Number(result)}
+import React, { useState, useEffect } from 'react';
+import NumberFormat from 'react-number-format';
+
+function Calculator() {
+  const [preState, setPreState] = useState('');
+  const [curState, setCurState] = useState('');
+  const [input, setInput] = useState('0');
+  const [operator, setOperator] = useState(null);
+  const [total, setTotal] = useState(false);
+
+  const inputNum = (e) => {
+    if (curState.includes('.') && e.target.innerText === '.') return;
+
+    if (total) {
+      setPreState('');
+    }
+    if (curState) {
+      setCurState((pre) => pre + e.target.innerText);
+    } else {
+      setCurState(e.target.innerText);
+      setTotal(false);
+    }
+  };
+
+  useEffect(() => {
+    setInput(curState);
+  }, [curState]);
+
+  useEffect(() => {
+    setInput('0');
+  }, []);
+
+  // equals
+
+  const equals = (e) => {
+    if (e?.target.innerText === '=') {
+      setTotal(true);
+    }
+    let cal;
+    switch (operator) {
+      case '÷':
+        cal = String(parseFloat(preState) / parseFloat(curState));
+        break;
+
+      case '+':
+        cal = String(parseFloat(preState) + parseFloat(curState));
+        break;
+      case 'X':
+        cal = String(parseFloat(preState) * parseFloat(curState));
+        break;
+      case '-':
+        cal = String(parseFloat(preState) - parseFloat(curState));
+        break;
+      default:
+        return;
+    }
+    setInput('');
+    setPreState(cal);
+    setCurState('');
+  };
+
+  const operatorType = (e) => {
+    setTotal(false);
+    setOperator(e.target.innerText);
+    if (curState === '') return;
+    if (preState !== '') {
+      equals();
+    } else {
+      setPreState(curState);
+      setCurState('');
+    }
+  };
+
+  const minusPlus = () => {
+    if (curState.charAt(0) === '-') {
+      setCurState(curState.substring(1));
+    } else {
+      setCurState(`-${curState}`);
+    }
+  };
+
+  const percent = () => {
+    if (preState) {
+      setCurState(String((parseFloat(curState) / 100) * preState));
+    } else {
+      setCurState(String(parseFloat(curState) / 100));
+    }
+  };
+
+  const reset = () => {
+    setInput('0');
+    setPreState('');
+    setCurState('');
+  };
```

Comment:
I have try before but  didn't working correctly because occurred  some  difficult linters issue
if  do you have time help me how I can fix it

Reply:
Hi @samiullahbahadur, since this is an optional requirement 🤷🏻, you could try this in your free time - refactor/play with code until you get results 😃. I totally understand the linter concerns 🤔.